transport/resolver.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2019 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transport
+
+import (
+	"net"
+	"net/url"
+)
+
+// Resolver is a function that resolves a url to a list of IP addresses.
+type Resolver func(url *url.URL) ([]net.Addr, error)
```

Comment:
`Resolver` conflicts with Go's standard [`net/Resolver`](https://pkg.go.dev/net#Resolver). Maybe a more specific name like `DNSResolver`? I'm also open to any other names.